app/middlewares/vendor_api_tracing_middleware.rb

```diff
@@ -0,0 +1,38 @@
+require 'redis'
+require './app/workers/vendor_api_trace_worker'
+
+class VendorAPITracingMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    started_at = Time.zone.now
+
+    @request = Rack::Request.new(env)
+    status, headers, response = @app.call(env)
+
+    begin
+      if vendor_api_path?
+        finished_at = Time.zone.now
```

Comment:
We might want to pass in the current time explicitly into the worker, otherwise the `created_at` will be misleading in case the worker retries.